### PR TITLE
JAMES-1902 reorganizes blob modules 

### PR DESCRIPTION
**Pull request by jeantil**
This PR also removes a couple duplicate dependencies that were present in the POM files.

### CODE REVIEW

File: server/container/guice/blob/blob-api-guice/pom.xml [hunk 1]
```diff
@@ -25,7 +25,7 @@
         <groupId>org.apache.james</groupId>
         <artifactId>james-server-guice</artifactId>
         <version>3.6.0-SNAPSHOT</version>
-        <relativePath>../pom.xml</relativePath>
+        <relativePath>../../pom.xml</relativePath>
```
Comment (chibenwa) [inline server/container/guice/blob/blob-api-guice/pom.xml:28]: Does it makes sense to keep the blob- prefix for folders inside a blob directory? Idem for the guice suffix.

Shouldn't the naming scheme be guice/blob/api ?

**Review by chibenwa — COMMENTED** [review]
That was a short follow up ;-)

We will launch tests on that tomorrow morning hanoi tz.

Thanks for your involvement.

File: server/container/guice/blob/blob-api-guice/pom.xml (hunk 1 above)
Comment (jeantil) [reply server/container/guice/blob/blob-api-guice/pom.xml:28]: you are right, I wanted to minimize changes but should have included that bit.

Comment (chibenwa) [reply server/container/guice/blob/blob-api-guice/pom.xml:28]: Should we do the same for the queue-xx-guice modules, if any?

Comment (jeantil) [reply server/container/guice/blob/blob-api-guice/pom.xml]: the queue-xxx- modules were already in *clean*  directories ( neither prefix nor suffix)

**Comment by jeantil** [comment]
Looking to make sure I had the proper sort order in the POM files I stumbled upon sortpom, wich led me to `site/src/xdoc/contribute` which helpfully provides: 
```
mvn com.github.ekryd.sortpom:sortpom-maven-plugin:sort -Dsort.keepBlankLines -Dsort.sortDependencies=groupId,artifactId -Dsort.nrOfIndentSpace=4 -Dsort.createBackupFile=false -Dsort.sortModules=true -Dsort.expandEmptyElements=false  -Dsort.predefinedSortOrder="recommended_2008_06"
```
I'm not sure if this recommandation is still applicable because running it returned over 150 incorrectly sorted pom, I applied the fixes to pom files from :
- `server/container/guice/queue/*`
- `server/container/guice/blob/*`
- `server/container/guice/pom.xml`

hopefully that's ok

**Comment by chibenwa** [comment]
We did not run that for ages, but the rules still applies, and we should do a run... (conflicts likely expected!) You are very right...

**Review by chibenwa — APPROVED** [review]
A well invested lunch pause!

**Comment by jeantil** [comment]
>We did not run that for ages, but the rules still applies, and we should do a run... (conflicts likely expected!) You are very right...

Should it be configured to run automatically on every compile ?
something like 
```
mvn compile -Psort
```
to make it easier to sort the modules ? then each new PR can commit the fixed poms it has modified

**Comment by chibenwa** [comment]
> Should it be configured to run automatically on every compile ?

And abort the build if needed? => :+1:

**Review by Arsnael — APPROVED** [review]
Nice work, thanks!

**Review by mbaechler — APPROVED** [review]

**Review by rouazana — APPROVED** [review]

**Comment by chibenwa** [comment]
I just merged this work. Can we close the PR?

**Pull request closed without merging**